src/map/script.cpp

```diff
@@ -3121,6 +3136,12 @@ void script_array_update(struct reg_db *src, int64 num, bool empty)
 int set_reg(struct script_state* st, struct map_session_data* sd, int64 num, const char* name, const void* value, struct reg_db *ref)
 {
 	char prefix = name[0];
+	size_t vlen = 0;
+	if ( !script_check_RegistryVariableLength(0,name,&vlen) )
+	{
+		ShowError("set_reg: Variable name length is too long (aid: %d, cid: %d): '%s' sz=%d\n", sd->status.account_id, sd->status.char_id, name, vlen);
```

Comment:
Will crash for non-player variables.

Reply:
why would it crash it just checking the len of a string.

Reply:
sd is used here without proper checking.

Reply:
Is `set_reg` ever called with a null session?

Reply:
judging by the code a few lines below without checking the source references i'd say: definitely.

Reply:
ohh I get it, you meant in the error msg, gotcha, yeah that annoying, I tough you where commenting in 
"if ( !script_check_RegistryVariableLength(0,name,&vlen) )"
K then should I remove the (aid: %d, cid: %d) part, or do you want that info if the sd is avalaible ?